### DISCUSSION TITLE
Distributed tempo on top of istio

### DISCUSSION THREAD

**Discussion by dvordrova** [grafana/tempo#3620] [Show and tell]
I'm probably bad at searching but it was kinda challenging to find how to run distributed tempo on top of istio

So I made an example how to deal with this one https://github.com/dvordrova/kind-istio-tempo-otel/tree/main that would be easy to setup

The main difference from default configuration that it has retries for memberlist joins and sets needed appProtocols for every service
Strange part was that the gossip-ring had to have appProtocol=tcp, despite it's targetPort named http-memberlist

Bonus picture, but I still can't make memberlist ring and memcached to appear on it
![tempo in tempo service graph](https://github.com/grafana/tempo/assets/3331351/5598ec5a-e332-4cef-93c0-2080974191b1)

**Comment by joe-elliott** [comment]
Nice! I have never tried to run Tempo on top of a service mesh. Appreciate the work and notes for the community.

**Comment by rodrigorodrigues** [comment]
Hi @dvordrova,

Do you have a sample for Service Graph with Istio/Ingress? Not sure what I'm doing wrong but the metrics doesn't show right for Service Graph.

<img width="1285" alt="image" src="https://github.com/user-attachments/assets/5eba3026-0575-455d-b759-a99a8455ab52" />

**Reply by dvordrova** [reply]
Hi, you can check out https://github.com/dvordrova/kind-istio-tempo-otel/tree/main README

So the crucial part is to make request to send some spans through istio

e.g. `curl http://localhost:30080/svc2/proxy`

**Reply by dvordrova** [reply]
If something doesn't work, I would probably debug it like this:
1) understand if I have spans generated
2) do I have sidecars/applications (that send spans) set up correctly (checking its logs and configurations)
3) tempo logs
4) do I have ingress involved

**Reply by joe-elliott** [reply]
Here are two related issues that may help understand what is happening:

https://github.com/grafana/tempo/issues/1418
https://github.com/grafana/tempo/issues/3414

**Reply by rodrigorodrigues** [reply]
Hi @joe-elliott thanks for those links, I think this can be fixed by  prometheus relabel_configs to _replace span_name with service_name_ but not quite sure how to use it.

In prometheus added the following.
```
  metric_relabel_configs:
  - source_labels: [service_name]
    separator: ;
    regex: (.*)
    target_label: span_name
    replacement: $1
    action: replace
```

My metrics look like this

```
traces_spanmetrics_calls_total{__metrics_gen_instance="tempo-0", 
service="authentication-service.default", 
service_name="authentication-service.default", 
source="tempo", 
span_kind="SPAN_KIND_SERVER", 
span_name="ingress", 
status_code="STATUS_CODE_UNSET"}
```
